src/BitStringAddresses/bosefs.jl

```diff
@@ -327,36 +325,60 @@ end
         result += bosons
         chunk >>>= bosons % UInt
     end
-    return (result, orbital, bit_position), (i, chunk, bits_left, orbital)
+    return BoseFSIndex(result, orbital, bit_position), (i, chunk, bits_left, orbital)
 end
 
-"""
-    BoseFS2C{NA,NB,M,AA,AB} <: AbstractFockAddress
+function find_site(b::BoseFS, index)
+    last_occnum = last_site = last_offset = 0
+    for (occnum, site, offset) in occupied_orbitals(b)
+        dist = index - site
+        if dist == 0
+            return BoseFSIndex(occnum, index, offset)
+        elseif dist < 0
+            return BoseFSIndex(0, index, offset + dist)
+        end
+        last_occnum = occnum
+        last_site = site
+        last_offset = offset
+    end
+    offset = last_offset + last_occnum + index - last_site
+    return BoseFSIndex(0, index, offset)
+end
 
-Address type that constructed with two [`BoseFS{N,M,S}`](@ref). It represents a
-Fock state with two components, e.g. two different species of bosons with particle
-number `NA` from species S and particle number `NB` from species B. The number of
-orbitals `M` is expected to be the same for both components.
-"""
-struct BoseFS2C{NA,NB,M,SA,SB} <: AbstractFockAddress
-    bsa::BoseFS{NA,M,SA}
-    bsb::BoseFS{NB,M,SB}
+function find_particle(b::BoseFS, index, n=1)
```

Comment:
Why do we need a `find_particle` function? Indexing or enumerating indistinguishable particles is a strange concept.
Doesn't this function really enumerate the modes with occupation > n?
Maybe we better call it 
```suggestion
function find_occupied_mode(b::BoseFS, index, n=1)
```